build/SharedFx.targets

```diff
@@ -121,6 +127,137 @@
       File="$(SharedFxPublishDirectory)\.version"
       Lines="@(VersionLines)"
       Overwrite="true" />
+  </Target>
+
+  <Target Name="PrepareForCrossGen" DependsOnTargets="PrepareForSharedFx;BuildSharedFx">
+    <PropertyGroup>
+      <RuntimePackageName>Microsoft.NETCore.App</RuntimePackageName>
+      <CrossGenTool>crossgen</CrossGenTool>
+      <CrossGenTool Condition="$([MSBuild]::IsOSPlatform('Windows'))">$(CrossGenTool).exe</CrossGenTool>
+
+      <LibPrefix Condition="$([MSBuild]::IsOSPlatform('Linux')) OR $([MSBuild]::IsOSPlatform('OSX'))">lib</LibPrefix>
+      <LibExtension>.so</LibExtension>
+      <LibExtension Condition="$([MSBuild]::IsOSPlatform('Windows'))">.dll</LibExtension>
+      <LibExtension Condition="$([MSBuild]::IsOSPlatform('OSX'))">.dylib</LibExtension>
+    </PropertyGroup>
+
+    <!-- Determine runtime location -->
+    <MSBuild Projects="$(SharedFxWorkDirectory)SharedFx.csproj" Targets="GetPackageDefinitions" >
+      <Output TaskParameter="TargetOutputs" ItemName="PackageDefinitions" />
+    </MSBuild>
+
+    <ItemGroup>
+      <RuntimePackage Include="@(PackageDefinitions)" Condition="$([System.String]::new('%(PackageDefinitions.Name)').Contains('runtime')) AND $([System.String]::new('%(PackageDefinitions.Name)').Contains('$(RuntimePackageName)'))" />
```

Comment:
Instead of batching, would this work? 
```xml
<RuntimePackage Include="@(PackageDefinitions->WithMetadataValue('Name', 'runtime.$(RuntimePackageName)'))" />
```

Reply:
it's a little harder to get this to work since the resolved package actually looks like runtime.win-x64.Microsoft.NETCore.App. It's also a little tricky to determine the actual runtime since we use the RID win7-x64 which resolves to win-x64 of the runtime package. IIRC we needed to use win7-x64 instead of win-x64 because some of our dependencies had specific native assets that needed to be picked.

Reply:
i see. Seems like it should be hard to establish this mapping using conditional.

My concern is that if the CoreCLR team introduces a new RID, we could end up pickup something we may not have meant to use.

Reply:
I think in this case, it should be okay. Since as far as I can tell, the package definitions I'm searching through only contain packages that are restored. Restore takes care of the RID resolution.